Remove _createDataStoreWithProps (#22996)
microsoft/FluidFramework: .changeset/gold-maps-cut.md

Reviewer: Can you also add a note at the bottom with a link to the release notes where these APIs were deprecated? Something like:

> The initial deprecations of the now changed or removed types were announced in Fluid Framework v2.4.0: [Several MergeTree Client Legacy APIs are now deprecated](https://github.com/microsoft/FluidFramework/blob/main/RELEASE_NOTES/2.4.0.md#several-mergetree-client-legacy-apis-are-now-deprecated-22629)

---
"@fluidframework/aqueduct": minor
"@fluidframework/container-runtime": minor
"@fluidframework/container-runtime-definitions": minor
"@fluidframework/datastore": minor
"@fluidframework/runtime-definitions": minor
"@fluidframework/test-runtime-utils": minor
---

> # Removed `_createDataStoreWithProps`

Reviewer: ```suggestion
# The createDataStoreWithProps APIs on ContainerRuntime and IContainerRuntimeBase have been removed
```

Removed two main APIs, `ContainerRuntime.createDataStoreWithProps` and `IContainerRuntimeBase.createDataStoreWithProps`

> has been removed.

Reviewer: The `ContainerRuntime.createDataStoreWithProps` and `IContainerRuntimeBase.createDataStoreWithProps` APIs have been removed.

> For alternative solutions use `PureDataObjectFactory.createInstanceWithDataStore` and pass in props via the `initialState`
> parameter.

Reviewer: ```suggestion
Replace uses of these APIs with `PureDataObjectFactory.createInstanceWithDataStore` and pass in props via the `initialState`
parameter.
```

---
"@fluidframework/aqueduct": minor
"@fluidframework/container-runtime": minor
"@fluidframework/container-runtime-definitions": minor
"@fluidframework/datastore": minor
"@fluidframework/runtime-definitions": minor

> "@fluidframework/test-runtime-utils": minor

Reviewer 2: @alexvy86 I guess we should only add package from the API is removed and not all packages which are in the PR?
cc: @tylerbutler

---

# The createDataStoreWithProps APIs on ContainerRuntime and IContainerRuntimeBase have been removed

Removed two main APIs, `ContainerRuntime.createDataStoreWithProps` and `IContainerRuntimeBase.createDataStoreWithProps`

> have been removed.

Reviewer 2: nit: This is redundant. Removed used twice.
